### ISSUE TITLE
Feature request: Inlined values

### ISSUE THREAD

**Issue by hlascelles** [roboll/helmfile#40]
Would there be any appetite to allow inlining values in their natural nested form (instead of name/value pair). I imagine it would require temporary file creation during execution, unless there are libs to convert nested yaml to json dot notation. Not sure if that is beyond helmfile's remit.

eg:

```
charts:
  - name: vault                      
    namespace: vault             
    chart: roboll/vault-secret-manager
    inlined_values: 
      some:
        quite:
        - name: deep
          value: 
            variable: {{ env "VAL" }}
```

**Comment by mumoshu** [comment]
@hlascelles Hi! Thanks for the request.

> I imagine it would require temporary file creation during execution, unless there are libs to convert nested yaml to json dot notation

Would it work for you if helmfile translated your example to:

```yaml
set:
- name: some.quite.deep
   value: "{{ env \"VAL\" }}"
```

Is this what you're looking for?

And also, I'm still unable see a concrete benefit from achieving what you suggest 🤔 

Just curious, but what's your goal of having the another way to specify deeply nested values?

**Comment by hlascelles** [comment]
You're right, there's nothing we can't do with setting name value pairs. I just realised that we had a large number of values files, some complex, some not.

Some were essentially one liners, eg heapster was just:
```
rbac:
  create: true
```

... whereas the complex ones also have ENV values that we need to pass in (so we're currently templating those in temporary files).

The heapster example could easily go into a name/value pair, but then it would be different to the complex charts which need to remain in nested form if we are to understand them properly. Or we split the complex values files to have name/value pairs for ENV *and* a complex values file.

I imagined being able to glance at one (large :smiley:) file, which would show all the values in their nested form, next to the chart that is consuming them, with ENV templating.

This is a "nice to have" (for us) feature request - we're having good success as it is.  Yes, if there is a "yaml to json dot notation" converter helmfile could use internally (post ENV templating), that would do the trick nicely.

**Comment by maver1ck** [comment]
I think the goal is to specify values in helmfile the same way we're specifying them in charts.

**Comment by mumoshu** [comment]
Hey @hlascelles @maver1ck, we have an another thread which is hot about (1) templating entire `values.yaml` before passing it to helm for processing and (2) implement/encourage inlining values into `values:` and prefer doing so over `set:` in `helmfile.yaml`

https://github.com/roboll/helmfile/issues/119#issuecomment-384830608

Would the discussed feature address your use-cases?

**Comment by mumoshu** [comment]
@hlascelles Exactly. Please look forward to it 👍

**Comment by mumoshu** [comment]
I believe this is resolved via #139. Please feel free to reopen or raise another issue if not!
